File: src/main/java/in/neuw/resource/demo/models/TestDto.java
package in.neuw.resource.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.http.HttpStatus;

/**
 * @author Karanbir Singh on 04/26/2020
 */
@Getter
@Setter
@Accessors(chain = true)
public class TestDto {

    private String message;

    @JsonIgnore
    private HttpStatus status;

    private int code;

}

File: src/main/kotlin/in/neuw/resource/demo/models/TestDto.kt
package `in`.neuw.resource.demo.models

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.http.HttpStatus

data class TestDto(var message: String = "") {

    @JsonIgnore
    var status:HttpStatus = HttpStatus.OK

    var code: Int = status.value()

}